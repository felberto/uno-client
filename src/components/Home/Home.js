import React, {Component} from 'react';
import logo from '../../logo.svg';
import {Button} from "react-bootstrap";
import CreateLobbyModal from '../Modal/CreateLobbyModal';
import JoinLobbyModal from "../Modal/JoinLobbyModal";

class Home extends Component {

    constructor(props, context) {
        super(props, context);

        this.state = {
            show: false
        };
    }

    render() {
        let createLobbyModalClose = () => this.setState({createLobbyModalShow: false});
        let joinLobbyModalClose = () => this.setState({joinLobbyModalShow: false});

        return (
            <div>
                <div style={{width: '100vh', height: '80vh', margin: '0 auto', padding: '10%'}}>
                    <div style={{
                        display: 'inline-block',
                        float: 'left',
                        width: '50%',
                        height: '100%',
                        backgroundColor: 'red'
                    }}>
                        <img src={logo} className="App-logo" alt="logo"/>
                    </div>
                    <div style={{
                        display: 'inline-block',
                        margin: '0 auto',
                        padding: '3px',
                        float: 'right',
                        width: '50%',
                        height: '100%',
                        backgroundColor: 'blue'
                    }}>
                        <Button onClick={() => this.setState({createLobbyModalShow: true})} variant="secondary"
                                size="lg"
                                active style={{display: 'block'}}>
                            Lobby erstellen
                        </Button>
                        <CreateLobbyModal
                            show={this.state.createLobbyModalShow}
                            onHide={createLobbyModalClose}
                        />
                        < br/>
                        <Button onClick={() => this.setState({joinLobbyModalShow: true})} variant="secondary"
                                size="lg"
                                active
                                style={{display: 'block'}}>
                            Lobby beitreten
                        </Button>
                        <JoinLobbyModal
                            show={this.state.joinLobbyModalShow}
                            onHide={joinLobbyModalClose}
                        />
                    </div>
                </div>
            </div>
        );
    }
}

export default Home;