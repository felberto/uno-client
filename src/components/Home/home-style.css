div.row {
    margin: 0;
}

.home-uno {
    margin: 0;
    font-size: 13em;
    color: #ffd400;
    text-shadow: -11px 11px 0 #000;
}
