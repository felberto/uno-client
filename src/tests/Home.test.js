import React from 'react';
import Home from "../components/Home/Home"
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

describe('load Home', () => {
    it('compare snapshot', () => {
        const tree = renderer.create(<BrowserRouter><Home/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});