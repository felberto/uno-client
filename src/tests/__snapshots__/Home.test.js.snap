// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load Home compare snapshot 1`] = `
<div
  className="background row"
  style={
    Object {
      "height": 768,
    }
  }
>
  <div
    className="col-md-6 center-horizontal-vertical"
  >
    <p
      className="home-uno"
    >
      UNO
    </p>
  </div>
  <div
    className="col-md-6 center-horizontal-vertical"
  >
    <button
      className="btn active btn-dark btn-lg"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "width": "50%",
        }
      }
      type="button"
    >
      Create Lobby
    </button>
    <br />
    <button
      className="btn active btn-dark btn-lg"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "width": "50%",
        }
      }
      type="button"
    >
      Join Lobby
    </button>
  </div>
</div>
`;
